larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewLongTracksTool.cc

```diff
@@ -0,0 +1,173 @@
+/**
+ *  @file   larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewLongTracksTool.cc
+ *
+ *  @brief  Implementation of the long tracks tool class.
+ *
+ *  $Log: $
+ */
+
+#include "Pandora/AlgorithmHeaders.h"
+
+#include "larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewLongTracksTool.h"
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+TwoViewLongTracksTool::TwoViewLongTracksTool() :
+    m_minMatchedFraction(0.3f),
+    m_minMatchingScore(0.95f),
+    m_minMatchedSamplingPoints(15),
+    m_minXOverlapFraction(0.9f),
+    m_minMatchedSamplingPointRatio(2)
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool TwoViewLongTracksTool::HasLongDirectConnections(IteratorList::const_iterator iIter, const IteratorList &iteratorList)
+{
+    for (IteratorList::const_iterator iIter2 = iteratorList.begin(), iIter2End = iteratorList.end(); iIter2 != iIter2End; ++iIter2)
+    {
+        if (iIter == iIter2)
+            continue;
+
+        if (((*iIter)->GetCluster1() == (*iIter2)->GetCluster1()) || ((*iIter)->GetCluster2() == (*iIter2)->GetCluster2()))
+            return true;
+    }
+
+    return false;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool TwoViewLongTracksTool::IsLongerThanDirectConnections(IteratorList::const_iterator iIter, const MatrixType::ElementList &elementList,
+    const unsigned int minMatchedSamplingPointRatio, const pandora::ClusterSet &usedClusters)
```

Comment:
Looking at where this function is called `minMatchedSamplingPointRatio` is always passed `m_minMatchedSamplingPointRatio`.
`minMatchedSamplingPointRatio` can probably be dropped as an argument and the function can always assess using `m_minMatchedSamplingPointRatio` instead.

Reply:
This function is static, so it cannot simply access `m_minMatchedSamplingPointRatio`. What do you suggest?

Reply:
I'd suggest leave this as is, especially as there's already a precedent for doing it this way in the three-view case.

I think that this was a member variable originally, but it became an argument when the function was made static (which was to enable reuse by another tool).

Reply:
Will leave it as it is then, thanks!